odoo/models.py

```diff
@@ -5726,6 +5726,14 @@ def exists(self) -> Self:
         new_ids, ids = partition(lambda i: isinstance(i, NewId), self._ids)
         if not ids:
             return self
+
+        # Special version without query
+        field_to_check = next((f for f in self._fields.values() if f.prefetch is True), None)
+        if field_to_check:
+            field_cache = self.env.cache._get_field_cache(self, field_to_check)
+            if all(_id in field_cache for _id in ids):
```

Comment:
A variable starting with `_` is meant to be unused.
```suggestion
            if all(id_ in field_cache for id_ in ids):
```